src/main/java/org/mbari/cthulhu/ui/player/MediaPlayerTimer.java

```diff
@@ -51,48 +51,77 @@
     /**
      * Captures system time when previousMpMillis is updated.
      */
-    private long previousSysMillis = 0;
+    private long previousSysMillisForMp = 0;
 
     /**
      * This is either +1 or -1 indicating time direction according to sign of
      * reported mediaPlayer time difference.
      */
     private long timeDirection;
-
+    
+    /**
+     * Was media playing in previous call to the timer?
+     */
+    private boolean wasPlaying = false;
+    
+    /**
+     * Captures system time when timer was called. Only actually used for logging, this
+     * helps visualize how accurate the frequency of the calls are wrt intended PERIOD.
+     */
+    private long previousSysMillis = 0;
+    
     /**
      * Create a timer.
      *
      * @param mediaPlayer media player that generates time-changed events
-     * @param onTick function to execute on each tick of the timer depending of time of previous call
+     * @param onTick function to execute on each tick of the timer
      */
     MediaPlayerTimer(MediaPlayer mediaPlayer, Consumer<Long> onTick) {
         timer.scheduleAtFixedRate(new TimerTask() {
             @Override
             public void run() {
+                final long newMpMillis = mediaPlayer.status().time();
+                
                 if (mediaPlayer.status().isPlaying()) {
-                    final long newMpMillis = mediaPlayer.status().time();
+                    wasPlaying = true;
+                    
+                    final long currSysMillis = System.currentTimeMillis();
```

Comment:
I did a little research to see tradeoffs between `currentTimeMillis()` and `nanoTime()`. The tl;dr is that the performance is comparible. So currentTimeMillis() is AOK to use here. Good write up on this at <http://pzemtsov.github.io/2017/07/23/the-slow-currenttimemillis.html>